customer-terminate-service/domain/src/test/java/se/inera/intyg/cts/domain/service/ExportPackageTest.java

```diff
@@ -0,0 +1,55 @@
+package se.inera.intyg.cts.domain.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static se.inera.intyg.cts.domain.util.TerminationTestDataFactory.defaultTermination;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import se.inera.intyg.cts.domain.model.Termination;
+import se.inera.intyg.cts.domain.model.TerminationStatus;
+import se.inera.intyg.cts.domain.repository.InMemoryTerminationRepository;
+
+class ExportPackageTest {
+
+  private InMemoryTerminationRepository inMemoryTerminationRepository;
+  private DummyCreatePackage dummyCreatePackage;
+  private ExportPackage exportPackage;
+  private Termination termination;
+
+  @BeforeEach
+  void setUp() {
+    termination = defaultTermination();
+    dummyCreatePackage = new DummyCreatePackage();
+    inMemoryTerminationRepository = new InMemoryTerminationRepository();
+    exportPackage = new ExportPackage(dummyCreatePackage, inMemoryTerminationRepository);
+  }
+
+  @Test
+  void shallGeneratePasswordForPackage() {
+    exportPackage.export(termination);
+
+    assertNotNull(dummyCreatePackage.password());
+  }
+
+  @Test
+  void shallStorePasswordForPackageInTermination() {
+    exportPackage.export(termination);
+
+    assertEquals(dummyCreatePackage.password(),
+        inMemoryTerminationRepository.findByTerminationId(termination.terminationId())
```

Comment:
The tests run an export and verifies that the termination.export() has received the generated password. Instead of using a mock and verifying its called arguments, the test can just check the termination in the repository.

Reply:
This test actually tests the implementation of findByTerminationId in InMemoryTerminationRepository in the assertEquals. What we realy want to test is that the steps in `exportPackage.export(termination);` has been made.